Develop

adamaltman opened the pull request.

garcijo committed:
Added vagrant to the project

The project now uses vagrant for easier development setup

garcijo committed:
Moved config settings to yaml

Configuration settings were moved from settings.php to
/environment/main.yaml

garcijo committed:
Composer PSR-4 class autoloading implemented

Classes are now loaded through composer's PSR-4 autoloader.

garcijo committed:
routes cleaned up and actions created

All the code in routes.php has been cleaned up and moved into separate
files in the Action folder.

adamaltman commented:
This PR is bad.

However, you need to improve your usage of git and github.  Create a feature branch (related to the issues I made), and then make a PR, and link it to the issue.  Then get it reviewed.  You shouldn't close the issue, when it's not closed.  It's closed when it is merged.

adamaltman commented:
You need to do a global git ignore on .vagrant and .idea because you have a lot of files that shouldn't be in here...

garcijo commented:
Ok, I'll start doing that right away. As for this pull request, should I merge it or should I ignore it and make a new one once I'm done with one of the issues?

adamaltman commented:
The proper thing to do is not to merge a bad PR... You should at a minimum remove the files that are here (.vagrant and .idea).

adamaltman commented on src/Web/Action/HomePageAction.php:
```diff
@@ -0,0 +1,45 @@
+            return $response;
+        }
+    }
+}
```
part of code standards is to have a blank line at the end of the file...  (you can make that a setting in phpstorm)

adamaltman commented on src/Web/Action/MusicPageAction.php:
```diff
@@ -0,0 +1,53 @@
+    public function __invoke(Request $request, Response $response, $args)
+    {
+        // Verify if user is authenticated. If false, redirect to login
+        if (isset($_SESSION['user'])) {
```
I think perhaps a perfect case for middleware -- is user authenticated?  If not, redirect to the login.

garcijo committed:
Removed vagrant files

aded .vagrant and .idea to gitignore to remove the files from the repo

adamaltman commented:
Nice work overall. :+1:

adamaltman commented:
I believe you have to proactively remove them from the project (once on the gitignore, any further changes won't be committed, but the original files are still there).  I think I would do something like: `git rm <file>`.

garcijo committed:
.idea and .vagrant fully removed

The last commit updated .gitignore but didn't remove the existing
.vagrant and .idea folders

garcijo commented:
This PR fixes issues #3: add namespacing and autoloading, #5: config settings to yaml, and #6: make actions
